src/controller/CHIPDeviceController.cpp

```diff
@@ -157,9 +159,11 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
         mStorageDelegate->SetStorageDelegate(this);
     }
 
-    mTransportMgr = chip::Platform::New<DeviceTransportMgr>();
-    mSessionMgr   = chip::Platform::New<SecureSessionMgr>();
-    mExchangeMgr  = chip::Platform::New<Messaging::ExchangeManager>();
+    mTransportMgr          = chip::Platform::New<DeviceTransportMgr>();
+    mSessionMgr            = chip::Platform::New<SecureSessionMgr>();
+    mExchangeMgr           = chip::Platform::New<Messaging::ExchangeManager>();
+    messageCounterManager  = chip::Platform::New<secure_channel::MessageCounterManager>();
```

Comment:
What's the benefit of assigning it to a local stack variable, and then transferring it to `mMessageCounterManager`?

Reply:
It hides implementation in the header file, but maybe it is not necessary.